src/common/paddle.ts

```diff
@@ -0,0 +1,31 @@
+import {
+  type Environment,
+  Paddle,
+  type Subscription,
+} from '@paddle/paddle-node-sdk';
+import { logger } from '../logger';
+
+export const paddleInstance = new Paddle(process.env.PADDLE_API_KEY, {
+  environment: process.env.PADDLE_ENVIRONMENT as Environment,
+});
+
+type CancelSubscriptionProps = {
+  subscriptionId: string;
+};
+export const cancelSubscription = async ({
+  subscriptionId,
+}: CancelSubscriptionProps): Promise<Subscription | undefined> => {
+  try {
+    return await paddleInstance.subscriptions.cancel(subscriptionId, {
+      effectiveFrom: null,
+    });
+  } catch (e) {
```

Comment:
Because user requested deletion, seemed wrong to "not to it" because let's say Paddle is down?
That why I added log so we can always retro do it?

But if you want we can throw, it will just be unclear on frontend why they didn't delete?

Reply:
yes it's better to stop the deletion, it's easy to forget to check these logs

Reply:
Alright, maybe we should then consider (non blocking) to show some toast or something if it fails.

Reply:
I'll throw manually, so we can keep the log anyway for digesting :)